lib/Sema/CSSimplify.cpp

```diff
@@ -1655,7 +1672,7 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
 
   if (matchFunctionRepresentations(func1->getExtInfo().getRepresentation(),
                                    func2->getExtInfo().getRepresentation(),
-                                   kind)) {
+                                   kind, getConstraintLocator(locator))) {
```

Comment:
Let's not allocate locator here but instead use `ConstraintLocatorBuilder` and check it like this:

```c++
auto last = locator.last();
if (!(last && last->is<LocatorPathElt::FunctionArgument>()))
   return;
```

in `matchFunctionRepresentations`. Safe us a few locations just for one check.